oauth2/oauth2-core/src/main/java/org/springframework/security/oauth2/core/endpoint/OAuth2AuthorizationRequest.java

```diff
@@ -50,6 +56,7 @@
 	private Set<String> scopes;
 	private String state;
 	private Map<String, Object> additionalParameters;
+	private URI authorizationRequestUri;
```

Comment:
Change to String for consistency.

Reply:
If we store this as a `String` than is it the encoded or decoded version? The `OAuth2AuthorizationRequestRedirectFilter` and `OAuth2AuthorizationRequestRedirectWebFilter` will need to know this before it initiates the redirect.